drivers/ethernet/eth_sam_gmac.c

```diff
@@ -2097,7 +2097,8 @@ static int ptp_clock_sam_gmac_adjust(struct device *dev, int increment)
 	Gmac *gmac = cfg->regs;
 	GMAC_TA_Type gmac_ta;
 
-	if ((increment <= -NSEC_PER_SEC) || (increment >= NSEC_PER_SEC)) {
+	if ((increment <= -(int)(NSEC_PER_SEC)) ||
```

Comment:
At the very least, please remove superfluous parens around `NSEC_PER_SEC`.

And the need to do this doesn't look pretty. I'd call all those guys who added that MISRA stuff to Zephyr, and asked: now what?

Reply:
Another option might be to remove that `U` suffix from NSEC_PER_SEC...

Reply:
Removing the U might also cause unintended results now. At least we can fix this here like suggested so that the comparison works in the future regardless of whether there the value is unsigned or not. Just fix the issue gitlint complains of and the extra () can be also removed, other than that looks ok.